File: backend/services/openrouter.py
import os
from dotenv import load_dotenv
from openai import OpenAI

load_dotenv()

client = OpenAI(
    base_url="https://openrouter.ai/api/v1",
    api_key=os.getenv("OPENROUTER_API_KEY")
)

MODEL = os.getenv(
    "MODEL",
    "deepseek/deepseek-chat-v3-0324:free"
)


def ask_ai(system_prompt, user_prompt):
    """
    Send a prompt to OpenRouter and return the AI response.
    """

    response = client.chat.completions.create(
        model=MODEL,
        messages=[
            {
                "role": "system",
                "content": system_prompt
            },
            {
                "role": "user",
                "content": user_prompt
            }
        ],
        temperature=0.7
    )

    return response.choices[0].message.content

File: backend/services/openrouter.js
const axios = require("axios");
require("dotenv").config();

const OPENROUTER_API_URL = "https://openrouter.ai/api/v1/chat/completions";

// Default model (change if you prefer another OpenRouter model)
const DEFAULT_MODEL =
  process.env.OPENROUTER_MODEL || "deepseek/deepseek-chat-v3-0324";

// ---------------------------
// Generic AI Request Function
// ---------------------------
async function generateAIResponse({
  systemPrompt,
  userPrompt,
  temperature = 0.7,
  maxTokens = 2000,
  model = DEFAULT_MODEL,
}) {
  try {
    const response = await axios.post(
      OPENROUTER_API_URL,
      {
        model,
        messages: [
          {
            role: "system",
            content: systemPrompt,
          },
          {
            role: "user",
            content: userPrompt,
          },
        ],
        temperature,
        max_tokens: maxTokens,
      },
      {
        headers: {
          Authorization: `Bearer ${process.env.OPENROUTER_API_KEY}`,
          "Content-Type": "application/json",

          // Optional but recommended
          "HTTP-Referer":
            process.env.APP_URL || "http://localhost:3000",

          "X-Title":
            process.env.APP_NAME || "AgentX Startup Builder",
        },
      }
    );

    return response.data.choices[0].message.content;
  } catch (error) {
    console.error("\n========== OPENROUTER ERROR ==========");

    if (error.response) {
      console.error(error.response.data);
    } else {
      console.error(error.message);
    }

    console.error("======================================\n");

    throw new Error("Unable to generate AI response.");
  }
}

// ---------------------------
// Agent Wrapper
// ---------------------------
async function runAgent(agentName, systemPrompt, userPrompt) {
  console.log(`Running ${agentName}...`);

  const result = await generateAIResponse({
    systemPrompt,
    userPrompt,
  });

  return {
    agent: agentName,
    result,
  };
}

module.exports = {
  generateAIResponse,
  runAgent,
};